src/ui.rs

```diff
@@ -250,8 +243,10 @@
             term.write_line("FAILED").expect("Failed to write terminal");
         }
 
-        rc.error = errors != 0;
-        rc
+        if errors != 0 {
+            return Err(anyhow!("Failed to run the target"));
```

Comment:
Nit: using `anyhow::bail` makes it a little less verbose:

```suggestion
            return bail!("Failed to run the target");
```